lib/presentation/extensions/send_file_extension.dart

```diff
@@ -80,18 +105,36 @@ extension SendFileExtension on Room {
         tempEncryptedFile.path,
         fileInfo.fileSize,
       );
+      if (thumbnail != null) {
+        encryptedThumbnail = await encryptedService.encryptFile(
+          fileInfo: thumbnail,
+          outputFile: tempEncryptedThumbnailFile,
+        );
+      }
     }
-    Uri? uploadResp;
+    Uri? uploadResp, thumbnailUploadResp;
 
     fakeImageEvent.rooms!.join!.values.first.timeline!.events!.first
         .unsigned![fileSendingStatusKey] = FileSendingStatus.uploading.name;
-    while (uploadResp == null) {
+    while (uploadResp == null ||
+        (encryptedThumbnail != null && thumbnailUploadResp == null)) {
       try {
         final uploadFileApi = getIt.get<UploadFileAPI>();
         final response = await uploadFileApi.uploadFile(fileInfo: tempfileInfo);
         if (response.contentUri != null) {
           uploadResp = Uri.parse(response.contentUri!);
         }
+        if (encryptedThumbnail != null && thumbnail != null) {
```

Comment:
what happen if we failed in uploading `original file`?

Reply:
it won't run upload thumbnail because it throws an exception before

Reply:
but the case is, `contentUri == null`? what happen?

Reply:
fixed